### ISSUE TITLE
Bash used for command line task on Windows in v2.165.2

### ISSUE THREAD

**Issue by mjvankampen** [microsoft/azure-pipelines-agent#2870]
[Worker_20200325-074832-utc.log](https://github.com/microsoft/azure-pipelines-agent/files/4379713/Worker_20200325-074832-utc.log)
## Having issue with Tasks?
Command line task not working but happens directly after update of the pipelines agent.

## Agent Version and Platform
Version: v2.165.2
Platform: Windows

## Azure DevOps Type and Version
Azure Devops https://dev.azure.com/Allseas

## What's not working?
All command line tasks fail as bash is used instead of cmd or powershell. Do note I only tried this within a docker container. So not outside a container.

```
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.164.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents:
C:\__w\1\s/conan/conan config install C:\__w\1\s/conan-config
========================== Starting Command Output ===========================
##[error]Unable to locate executable file: 'bash'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.
```

## Agent and Worker's Diagnostic Logs
Added log

**Comment by mjvk** [comment]
Using the pwsh task does work which is my fix for this.

**Comment by mjvk** [comment]
#2864 is the same issue. I think it should not be closed, either this issue should be fixed or the documentation should be updated. Now I had multiple people in my team running into this issue and spending time on it.

**Comment by jtpetty** [comment]
The issue is not that bash is being used instead of PowerShell to run the script specified in the task, but how we invoke the CommandLine task itself. Our task model supports running tasks via PowerShell or nodejs. The problem we were trying to address is that newer versions of Windows containers come with PowerShell Core, not PowerShell, which has different arguments and settings. This was causing the CommandLine task to fail on Windows containers without PowerShell. We updated the agent to prefer using the nodejs handler when running tasks in a container. It seems that handler has a dependency on running bash.

The workaround is to set ```agent.preferPowerShellOnContainers = true``` 

@damccorm - Would you be able to investigate why there is a dependency on bash in the nodejs handler? If we cannot remove it, we may need to reverse the default value of the variable mentioned above.

**Comment by mback2k** [comment]
My understand is that this was fixed by #2913.
